client/app/lib/components/table/builder/TableTemplate.ts

```diff
@@ -0,0 +1,28 @@
+import ColumnTemplate, { Data } from './ColumnTemplate';
+import {
+  CsvDownloadTemplate,
+  FilterTemplate,
+  IndexingTemplate,
+  PaginationTemplate,
+  SearchTemplate,
+  SortTemplate,
+  ToolbarTemplate,
+} from './featureTemplates';
+
+interface TableTemplate<D extends Data> {
+  data: D[];
+  columns: ColumnTemplate<D>[];
+  getRowId: (datum: D) => string;
```

Comment:
Because TanStack Table's `useReactTable`'s `getRowId` needs a `string` ID. I believe this ID is used as `key` for the React components, and component keys are supposed to be strings.